Client/mods/deathmatch/logic/CClientRadarMarkerManager.cpp

```diff
@@ -81,14 +81,16 @@ void CClientRadarMarkerManager::RemoveFromList(CClientRadarMarker* pMarker)
 
 void CClientRadarMarkerManager::SetDimension(unsigned short usDimension)
 {
-    m_usDimension = usDimension;
-    m_bOrderOnPulse = true;
+    if (m_usDimension != usDimension)
+        m_usDimension = usDimension;
+        m_bOrderOnPulse = true;
 }
 
 void CClientRadarMarkerManager::SetInterior(unsigned char ucInterior)
 {
-    m_ucInterior = ucInterior;
-    m_bOrderOnPulse = true;
+    if (m_ucInterior != ucInterior)
+        m_ucInterior = ucInterior;
+        m_bOrderOnPulse = true;
```

Comment:
And here too

Reply:
`m_bOrderOnPulse = true;` will be executed regardless of the condition.
Do it in this way:
```lua
if (m_ucInterior != ucInterior)
{
         m_ucInterior = ucInterior;
         m_bOrderOnPulse = true;
}
```